{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "from glob import glob\n",
    "import pandas as pd\n",
    "path = '/workdir/important_datasets/EPPO Dataframes/Kingdom/'\n",
    "files = glob(path + '*.gzip')\n",
    "dfs = pd.concat([pd.read_parquet(f) for f in files])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0        https://gd.eppo.int/taxon/TSWV00/datasheet\n",
       "1        https://gd.eppo.int/taxon/INSV00/datasheet\n",
       "2        https://gd.eppo.int/taxon/TRSV00/datasheet\n",
       "3        https://gd.eppo.int/taxon/TOCV00/datasheet\n",
       "4        https://gd.eppo.int/taxon/TYLCV0/datasheet\n",
       "                            ...                    \n",
       "4193      https://gd.eppo.int/taxon/AREAB/datasheet\n",
       "25966     https://gd.eppo.int/taxon/HKASE/datasheet\n",
       "24710     https://gd.eppo.int/taxon/GYNSP/datasheet\n",
       "18932     https://gd.eppo.int/taxon/EHRCA/datasheet\n",
       "21390     https://gd.eppo.int/taxon/EUPAD/datasheet\n",
       "Name: Datasheet, Length: 354, dtype: object"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfs = dfs.dropna(subset=['Datasheet'])    \n",
    "dfs.Datasheet"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 163,
   "metadata": {},
   "outputs": [],
   "source": [
    "\"\"\"\n",
    "scrape urls in dfs.Datasheet and parse 'datasheet' div class, put the content in a nested dict of this type {<h3>:[<h4>:<p>]}. save the dict in a new dict with the second last word of the url as key and the dict as value \n",
    "Example:\n",
    "\"\"\"\n",
    "a = \"\"\"\n",
    "<h1 class=\"title\">EPPO Datasheet: <i>Palm lethal yellowing type syndromes</i></h1>\n",
    "<!-- -->\n",
    "<h3>IDENTITY</h3>\n",
    "<div class=\"para0\">\n",
    "    <strong>Preferred name:</strong> <i>Palm lethal yellowing type syndromes</i><br>\n",
    "        <strong>Taxonomic position: </strong>Bacteria: Tenericutes: Mollicutes: Acholeplasmatales: Acholeplasmataceae<br>\n",
    "\t<strong>Other scientific names: </strong> <i>Coconut lethal yellowing phytoplasma</i>, <i>Palm lethal yellowing disease</i><br>\n",
    "\t<strong>Common names in English: </strong> Awka wilt disease, Bogia coconut syndrome, Cape St Paul wilt of coconut, Palm lethal yellowing disease, bronze leaf wilt of coconut (JM), kaincope disease of coconut, kribi disease of coconut, lethal yellowing of coconut, lethal yellowing syndromes of coconut, lethal yellowing type syndromes<br>\n",
    "<a href=\"https://gd.eppo.int/taxon/PHYP56/\">view more common names online...</a><br>\n",
    "<!-- -->\n",
    "<strong>Notes on taxonomy and nomenclature</strong>\n",
    "<div class=\"para0\">\n",
    "<p style=\"text-align: justify;\">Lethal yellowing disease of coconut was first observed in the Caribbean in the 19<sup>th</sup> century. However, it was only in 1972 that it was confirmed to be caused by phytoplasmas (Beakbane <em>et al.,</em> 1972; Plavsic-Banjac <em>et al.,</em> 1972). The many common names cited above, often from the affected localities, have been used for diseases of coconuts which are all now considered to be caused by different phytoplasmas associated with common syndromes: Lethal Yellowing Type Syndromes. These diseases differ mainly in their geographic distributions. Their host ranges all contain plants from the Arecacae family, including the coconut tree (<em>Cocos nucifera</em>). Many coconut diseases of unknown aetiology are described in the older literature and records of these have been appraised by Howard (1983). The name ‘bronze leaf wilt’ has been used for lethal yellowing in Jamaica but corresponds to a different disease in South America.</p><p style=\"text-align: justify;\">Because no complete genome sequences are currently available for any of these phytoplasmas, their taxonomy is based on 16S rRNA sequences, and two parallel classification systems have been developed, the 16Sr group system, based on restriction enzyme digest profiles of the 16S rDNA, and the <em>‘Candidatus</em> Phytoplasma’ species system, in which phytoplasmas sharing less than 97.5% similarity of their 16S rRNA gene sequence may be ascribed to different ‘<em>Ca</em>. Phytoplasma’ species when they are characterised by distinctive biological, phytopathological and genetic properties (Bertaccini <em>et al</em>., 2022, EFSA, 2017).</p><p style=\"text-align: justify;\">Phytoplasmas associated with palm lethal yellowing type syndromes are placed in the 16SrIV and 16SrXXII groups, both of which are divided in different subgroups reflecting the restriction profile of the 16SrRNA gene (Gurr <em>et al</em>., 2016; Wei <em>et al</em>., 2007).</p><p style=\"text-align: justify;\">In the framework of this datasheet, the following ‘<em>Candidatus</em> Phytoplasma’ species are considered to be associated with palm lethal yellowing type syndromes:</p><ul><li>‘<em>Candidatus</em> Phytoplasma aculeata’</li><li>‘<em>Candidatus</em> Phytoplasma cocostanzaniae’</li><li>‘<em>Candidatus</em> Phytoplasma dypsidis'</li><li><em>'Candidatus</em> Phytoplasma hispanola'</li><li><em>'Candidatus</em> Phytoplasma noviguineense'</li><li>‘<em>Candidatus</em> Phytoplasma palmae’</li><li>‘<em>Candidatus</em> Phytoplasma palmicola’</li><li><br></li></ul></div>\n",
    "\t<strong>EPPO Categorization:</strong> A1 list<br>\n",
    "\t<strong>EU Categorization:</strong> A1 Quarantine pest (Annex II A)<br>\n",
    "\t  <a href=\"https://gd.eppo.int/taxon/PHYP56/categorization\">view more categorizations online...</a><br>\n",
    "\n",
    "  <strong>EPPO Code:</strong> PHYP56</div>\n",
    "\n",
    "    <!-- -->\n",
    "<a class=\"btntitle\" href=\"#\"><h3><i class=\"fa fa-plus-circle\" aria-hidden=\"true\"></i> HOSTS<small class=\"pull-right\" title=\"Last modification\"><i class=\"fa fa-clock-o\" aria-hidden=\"true\"></i> 2024-06-18</small></h3></a>\n",
    "<div class=\"para1\" style=\"display: none;\">\n",
    "<p style=\"text-align: justify;\">The main host is coconuts (<em>Cocos nucifera</em>) but the lethal yellowing type syndromes have also been found on many different palms (Arecaceae)<em>.</em> The occurrence of symptoms similar to lethal yellowing together with the presence of phytoplasmas among collections of palms in affected areas in Florida (USA) indicates that more than 30 species of palms are susceptible to infection. Palm lethal yellowing type syndrome phytoplasmas have been detected in non Arecaceae plants, including different grasses in plots where palms were infested (Arocha-Rosete <em>et al.</em>, 2016; Gurr <em>et al</em>., 2016). The relevance of these plants as potential reservoirs of the phytoplasmas is not clear (EFSA, 2017). <em>Ca</em> ‘Phytoplasma noviguineense’ also includes banana (<em>Musa</em> sp. ) as another economically important host crop.</p><p>A detailed list of host plants for each phytoplasma species is available in EPPO Global Database. A global host list for all host plants of palm lethal yellowing type syndromes phytoplasmas is as follows:</p><strong>Host list:</strong> <i>Acrocomia aculeata</i>, <i>Adonidia merrillii</i>, <i>Aiphanes horrida</i>, <i>Aiphanes minima</i>, <i>Allagoptera arenaria</i>, <i>Areca catechu</i>, <i>Arecaceae</i>, <i>Arenga engleri</i>, <i>Arenga pinnata</i>, <i>Arenga</i>, <i>Attalea butyracea</i>, <i>Bismarckia nobilis</i>, <i>Borassus flabellifer</i>, <i>Brahea brandegeei</i>, <i>Butia odorata</i>, <i>Carludovica palmata</i>, <i>Carpentaria acuminata</i>, <i>Caryota mitis</i>, <i>Caryota rumphiana</i>, <i>Caryota urens</i>, <i>Chelyocarpus chuco</i>, <i>Chrysalidocarpus cabadae</i>, <i>Chrysalidocarpus decaryi</i>, <i>Chrysalidocarpus leptocheilos</i>, <i>Chrysalidocarpus lutescens</i>, <i>Cleome rutidosperma</i>, <i>Cocos nucifera</i>, <i>Cocos</i>, <i>Corypha umbraculifera</i>, <i>Corypha utan</i>, <i>Corypha</i>, <i>Cryosophila warscewiczii</i>, <i>Cyanthillium cinereum</i>, <i>Dictyosperma album</i>, <i>Dictyosperma</i>, <i>Elaeis guineensis</i>, <i>Gaussia attenuata</i>, <i>Gaussia</i>, <i>Howea belmoreana</i>, <i>Howea forsteriana</i>, <i>Hyophorbe verschaffeltii</i>, <i>Hyophorbe</i>, <i>Latania lontaroides</i>, <i>Livistona australis</i>, <i>Livistona chinensis</i>, <i>Macroptilium lathyroides</i>, <i>Nannorrhops ritchieana</i>, <i>Phoenix canariensis</i>, <i>Phoenix dactylifera</i>, <i>Phoenix reclinata</i>, <i>Phoenix roebelenii</i>, <i>Phoenix sylvestris</i>, <i>Phoenix</i>, <i>Pritchardia pacifica</i>, <i>Pritchardia thurstonii</i>, <i>Pseudophoenix sargentii</i>, <i>Ravenea rivularis</i>, <i>Roystonea regia</i>, <i>Sabal mexicana</i>, <i>Sabal palmetto</i>, <i>Saribus rotundifolius</i>, <i>Stachytarpheta jamaicensis</i>, <i>Syagrus romanzoffiana</i>, <i>Syagrus schizophylla</i>, <i>Syagrus sp.</i>, <i>Synedrella nodiflora</i>, <i>Trachycarpus fortunei</i>, <i>Veitchia arecina</i>, <i>Washingtonia robusta</i>\n",
    "</div>\n",
    "<!-- -->\n",
    "<a class=\"btntitle\" href=\"#\"><h3><i class=\"fa fa-plus-circle\" aria-hidden=\"true\"></i> GEOGRAPHICAL DISTRIBUTION<small class=\"pull-right\" title=\"Last modification\"><i class=\"fa fa-clock-o\" aria-hidden=\"true\"></i> 2024-06-18</small></h3></a>\n",
    "<div class=\"para1\" \"=\"\">\n",
    "<p style=\"text-align: justify;\">Lethal yellowing type syndromes are widely distributed in the intertropical zone. They are present in North and Central America and the Caribbean (‘<em>Ca</em>. Phytoplasma palmae’ and ‘<em>Ca</em>. Phytoplasma aculeata’ and '<em>Ca</em>. Phytoplasma hispanola'), in Africa (‘<em>Ca</em>. Phytoplasma palmicola’ and ‘<em>Ca</em>. Phytoplasma cocostanzaniae’ and in Oceania (<em>'Ca</em>. Phytoplasma noviguineense'<u>,</u> and ‘<em>Ca</em>. Phytoplasma dypsidis').</p><p>Distribution maps for each phytoplasma species is available in EPPO Global Database. The global distribution of lethal yellowing type syndromes of palms is as follows:</p><img class=\"img-responsive\" src=\"/taxon/PHYP56/download/distributionmap_png\" style=\"margin:10px 0px 10px 0px\">\n",
    "<strong>Africa: </strong>Cameroon, Cote d'Ivoire, Equatorial Guinea, Ghana, Kenya, Madagascar, Mozambique, Nigeria, Tanzania, Togo<br>\n",
    "<strong>North America: </strong>Mexico, United States of America (Florida, Louisiana, Texas)<br>\n",
    "<strong>Central America and Caribbean: </strong>Antigua and Barbuda, Belize, Cayman Islands, Cuba, Dominican Republic, Guadeloupe, Guatemala, Haiti, Honduras, Jamaica, Netherlands Antilles, St Kitts-Nevis<br>\n",
    "<strong>Oceania: </strong>Australia (Queensland), Papua New Guinea<br>\n",
    "</div>\n",
    "<!-- -->\n",
    "<!-- -->\n",
    "<a class=\"btntitle\" href=\"#\"><h3><i class=\"fa fa-plus-circle\" aria-hidden=\"true\"></i> BIOLOGY<small class=\"pull-right\" title=\"Last modification\"><i class=\"fa fa-clock-o\" aria-hidden=\"true\"></i> 2024-06-18</small></h3></a>\n",
    "<div class=\"para1\">\n",
    "<p style=\"text-align: justify;\">Palm lethal yellowing type syndromes are associated with phytoplasmas. Mycoplasma-like particles have been found in the sieve-tube elements of the phloem of coconut and other palms exhibiting characteristic symptoms (Beakbane <em>et al.,</em> 1972; Plavsic-Banjac <em>et al.,</em> 1972). Oxytetracycline treatment causes remission of disease symptoms (McCoy, 1975). Since then, the techniques of molecular biology and DNA sequencing have made it possible to show the association of phytoplasmas with these syndromes (Gurr <em>et al.</em>, 2016; Jones <em>et al</em>., 2021; Miyazaki <em>et al</em>., 2018; Soto <em>et al</em>., 2021).</p><p style=\"text-align: justify;\">Marked differences in susceptibility to the phytoplasmas are reported in different varieties of coconut palms as well as between other palm species. Susceptibility also varies with the phytoplasma species involved (Howard <em>et al</em>., 1979; Gurr <em>et al</em>., 2016; Palma-Cancino <em>et al.</em>, 2023).</p><p style=\"text-align: justify;\">To date, the only insect identified as a vector of a palm lethal yellowing type syndromes -associated phytoplasma is <em>Haplaxius crudus</em> in the Caribbean. <em>H. crudus</em> feeds abundantly on coconut leaves and population densities of <em>H. crudus</em> were observed to be up to 40 times higher in areas affected by palm lethal yellowing type syndromes than unaffected ones (Howard, 1980; Howard <em>et al</em>., 1983). Since then, evidence for <em>H. crudus</em> carrying phytoplasmas has been demonstrated (Dzido <em>et al</em>., 2020; Mou <em>et al</em>., 2022a; EPPO, 2024). The geographical distribution of <em>H</em><em>. crudus</em> in the Americas more or less coincides with the known distribution of the phytoplasmas (Howard, 1983). In areas other than the Americas, no vector has been positively identified. Other species of insects have been suggested as potential vectors (Brown <em>et al</em>., 2006; Bila <em>et al</em>., 2017; Fernández-Barrera <em>et al.,</em> 2022; Kwadjo <em>et al</em>., 2018; Lu <em>et al</em>., 2016; Mpunami <em>et al</em>., 2000; Ramos Hernández <em>et al.</em>, 2020).</p><p style=\"text-align: justify;\">Purcell (1985) noted that <em>H. crudus</em> is a very inefficient vector of lethal yellowing, but can be so abundant that a very low transmission rate is sufficient to spread the disease.</p><p style=\"text-align: justify;\">Oropeza <em>et al</em>. (2017) demonstrated the transmission of lethal yellowing phytoplasmas from coconut embryos to plantlets. However the relevance of seed transmission for disease spread is not clear.</p></div>\n",
    "<!-- -->\n",
    "<a class=\"btntitle\" href=\"#\"><h3><i class=\"fa fa-minus-circle\" aria-hidden=\"true\"></i> DETECTION AND IDENTIFICATION<small class=\"pull-right\" title=\"Last modification\"><i class=\"fa fa-clock-o\" aria-hidden=\"true\"></i> 2024-06-18</small></h3></a>\n",
    "<div class=\"para1\" style=\"display: block;\">\n",
    "<h4><strong>Symptoms</strong></h4><h4 style=\"text-align: justify;\"><strong>Morphology</strong></h4><p style=\"text-align: justify;\">Typical phytoplasma particles were found in sieve tubes of infected plants. They were ovoid, elongated and filamentous in shape and were bounded by a triple-layered structure comprising two electron-dense layers with a transparent layer between them (Plavsic-Banjac <em>et al.,</em> 1972).</p><h4 style=\"text-align: justify;\"><strong>Detection and inspection methods</strong></h4><p style=\"text-align: justify;\">In general, an early symptom is the drying up of developing inflorescences, and the premature drop of most or all fruits or seeds within a few days. In coconut palms the spathes enclosing the flowers become discoloured and the tips blacken. The youngest leaves next to the buds show water-soaked streaks which spread until there is a terminal rot of the growing point. Following the first symptoms there is progressive leaf discoloration, beginning with the older leaves and spreading rapidly to the younger ones. The foliage turns light-yellow and eventually orange-yellow. This symptom coincides with the death of root tips. Death occurs in <em>C. nucifera</em> about 4 months after the initial symptoms appear.</p><p style=\"text-align: justify;\">Several molecular methods exist to detect lethal yellowing type syndrome associated phytoplasmas both in plants and in vectors (Bahder <em>et al.</em>, 2018; Bahder <em>et al</em>., 2019; Christensen <em>et al</em>., 2004; Córdova <em>et al.</em>, 2014; Hodgetts <em>et al</em>., 2009; Lu <em>et al.</em>, 2016; Pilet, 2021). LAMP detection systems have also been developed to rapidly detect the phytoplasmas in the field (Dickinson, 2015; Tropicsafe, 2021a; Yu <em>et al</em>., 2023).</p><p style=\"text-align: justify;\">Sampling is most reliable when done in immature leaves taken from around the apical meristem, or inflorescences, which are rich in phloem (Harrison <em>et al.</em>, 1999). However, once plants are symptomatic, PCR testing of the phloem from the palm trunk (by drilling a hole 10–15 cm into the trunk and collecting sawdust) is a non-destructive method of successful phytoplasma detection (Harrison <em>et al.</em>, 2002). The drill should be sterilised (e.g. in 70% ethanol) prior to each sample collection.</p></div>\n",
    "<!-- -->\n",
    "<a class=\"btntitle\" href=\"#\"><h3><i class=\"fa fa-plus-circle\" aria-hidden=\"true\"></i> PATHWAYS FOR MOVEMENT<small class=\"pull-right\" title=\"Last modification\"><i class=\"fa fa-clock-o\" aria-hidden=\"true\"></i> 2024-06-18</small></h3></a>\n",
    "<div class=\"para1\">\n",
    "<p style=\"text-align: justify;\">Natural spread results from the movement of the known vector <em>H. crudus</em>, as well as of yet unknown vectors. Infected plants for planting, including ornamental species, can carry the pathogen in international trade. They can also carry infectious vectors, such as adults of <em>Haplaxius crudus</em> that are known to remain within palm foliage when plants are uprooted and transported to new localities. Ogle and Harries (2005) considered that the most likely means of transmission of the disease between Caribbean islands has been by the unintentional introduction of infected vectors on pasture grasses or animal fodder. Nymphs of <em>H. crudus</em> could also be moved in soil accompanying plants for planting or turf, but nymphs are not infected by the phytoplasmas.</p></div>\n",
    "<!-- -->\n",
    "<a class=\"btntitle\" href=\"#\"><h3><i class=\"fa fa-plus-circle\" aria-hidden=\"true\"></i> PEST SIGNIFICANCE<small class=\"pull-right\" title=\"Last modification\"><i class=\"fa fa-clock-o\" aria-hidden=\"true\"></i> 2024-06-18</small></h3></a>\n",
    "<div class=\"para1\">\n",
    "<h4><strong>Economic impact</strong></h4><p style=\"text-align: justify;\">Palm lethal yellowing phytoplasmas are a serious economic threat for coconut, causing their premature decline and death. Experience in Jamaica suggests that almost total destruction of a population of susceptible palms can occur: by 1979, an estimated 4 million coconut palms had been killed by palm lethal yellowing disease on the island. In Florida, out of an estimated 1-1.5 million <em>C. nucifera</em> on the mainland, 300 000 had died by 1983. In the case of Florida, not a coconut-producing area, the socio-economic loss has been as a result of a destruction of an important and valuable feature of the amenity vegetation. The disease is still causing a large impact nowadays in the Caribbean and Central America (Gurr <em>et al</em>., 2016; Mou <em>et al.</em>, 2022b), as well as in Africa (Bertaccini <em>et al</em>., 2023; Tropicsafe 2021b; Mpunami <em>et al</em>., 1999) and Papua New Guinea (Miyazaki <em>et al.</em>, 2018).</p><h4 style=\"text-align: justify;\"><strong>Control</strong></h4><p style=\"text-align: justify;\">Management of the disease relies on several steps: early identification of infected plants (based on symptoms and ideally confirmed by testing), removal of infected plants, control of the vectors, planting of healthy plants, weed control.</p><p style=\"text-align: justify;\">Research is being conducted to identify resistant genotypes (Gurr <em>et al</em>., 2016; Tropicsafe, 2021c).</p><p style=\"text-align: justify;\">Control of the vector <em>H. crudus</em> with insecticides by spraying or trunk injection is used in some crops but has not been economically successful in coconut plantations (Gurr <em>et al</em>., 2016). Management of the grass populations in coconut plantations has also been suggested as a means of control, since some grass species are hosts for the nymphs of <em>H. crudus</em> (although they are relatively poor hosts) (Howard, 1990).</p><p style=\"text-align: justify;\">Treatment of infected plants by injection into the trunk of antibiotics (e.g. oxytetracycline) had been shown to suppress the symptoms of lethal yellowing (McCoy, 1975) but was not considered sustainable as it should be applied at 4-monthly intervals and the use of antibiotics in agriculture is banned in many countries worldwide.</p><h4 style=\"text-align: justify;\"><strong>Phytosanitary risk</strong></h4><p style=\"text-align: justify;\">In the EPPO region, the most important economic crop at risk are date palms (<em>Phoenix dactylifera</em>) that are widely cultivated in Algeria, Morocco and Tunisia for fruit production. In addition, many palm species are used as ornamental trees in the entire EPPO region (indoor and outdoor). Movement of ornamental palms from infested areas could be just as hazardous to date-growing countries as the movement of date palms themselves, because of the possibility of spread from ornamental to date palms by existing auchenorrhynchan insects that may become local vectors.</p><p style=\"text-align: justify;\">The disease introduction into the EPPO region could also impact ornamental palms of considerable amenity value grown outside in the southern EPPO region and as well as valuable collections of palms under glass in botanic gardens, etc. In addition, palms are of great significance in nearly all modern indoor landscape plantings and their value to the horticultural trade is considerable in particular for mature plants.</p></div>\n",
    "<!-- -->\n",
    "<!-- -->\n",
    "<a class=\"btntitle\" href=\"#\"><h3><i class=\"fa fa-plus-circle\" aria-hidden=\"true\"></i> PHYTOSANITARY MEASURES<small class=\"pull-right\" title=\"Last modification\"><i class=\"fa fa-clock-o\" aria-hidden=\"true\"></i> 2024-06-18</small></h3></a>\n",
    "<div class=\"para1\">\n",
    "<p style=\"text-align: justify;\">In order to prevent the introduction of the disease to the EPPO region, the importation of host plants for planting, originating from areas where the disease occurs should be prohibited. Alternatively, measures such as pest-free areas may be appropriate, as is the case for similar pests, as well as pest-free production sites or pest-free places of production with conditions also ensuring the absence of vectors. Healthy planting material of host plants can be produced in the framework of a certification scheme. Post-entry quarantine (including testing) could be used in special cases.</p></div>\n",
    "<!-- -->\n",
    "<!-- -->\n",
    "<a class=\"btntitle\" href=\"#\"><h3><i class=\"fa fa-plus-circle\" aria-hidden=\"true\"></i> REFERENCES<small class=\"pull-right\" title=\"Last modification\"><i class=\"fa fa-clock-o\" aria-hidden=\"true\"></i> 2024-06-18</small></h3></a>\n",
    "<div class=\"para1\">\n",
    "<p>Arocha-Rosete Y, Diallo HA, Konan-Konan JL, Kouame AEP, Seka K, Kra KD, Toualy MN, Kwadjo KE, Daramcoum WAMP, Beugré NI, Ouattara BWM, Kouadjo Zaka CG, Allou K, Fursy-Rodelec ND, Doudjo-Ouattara ON, Yankey N, Dery S, Maharaj A, Saleh M, Summerbell R, Contaldo N, Paltrinieri S, Bertaccini A &amp; Scott J (2016) Detection and identification of the coconut lethal yellowing phytoplasma in weeds growing in coconut farms in Cote d’Ivoire. <em>Canadian Journal of Botany</em> <strong>38</strong>, 164–173</p><p>Bahder BW, Soto N, Komondy L, Mou DF, Humphries AR &amp; Helmick EE (2019) Detection and quantification of the 16SrIV-D phytoplasma in leaf tissue of common ornamental palm species in Florida using qPCR and dPCR. <em>Plant Disease</em> <strong>103</strong>(8), 1918-1922.</p><p>Bahder BW, Helmick EE, Mou DF, Harrison NA &amp; Davis R (2018) Digital PCR technology for detection of palm-infecting phytoplasmas belonging to group 16SrIV that occur in Florida. <em>Plant Disease</em> <strong>102</strong>(5), 1008-1014.</p><p>Beakbane AB, Slater CHW, &amp; Posnette AF (1972) Mycoplasmas in the phloem of coconut, <em>Cocos nucifera</em> L. with lethal yellowing disease. <em>Journal of Horticultural Science&nbsp;</em><strong>47</strong>, 256.</p><p>Bertaccini A, Arocha-Rosete Y, Contaldo N, Duduk B, Fiore N, Montano HG, Kube M, Kuo CH, Martini M, Oshima K &amp; Quaglino F (2022) Revision of the ‘<em>Candidatus</em> Phytoplasma’ species description guidelines. <em>International Journal of Systematic and Evolutionary Microbiology&nbsp;</em><strong>72</strong>(4), 005353.</p><p>Bertaccini A, Contaldo N, Feduzi G, Andeme AM, Yankey EN &amp; Rovesti L (2023) Molecular identification of ‘<em>Candidatus</em> Phytoplasma palmicola’ associated with coconut lethal yellowing in Equatorial Guinea. <em>Annals of Applied Biology</em> <strong>183</strong>(3), 262-270. <a href=\"https://doi.org/10.1111/aab.12854\" rel=\"noopener noreferrer\" target=\"_blank\">https://doi.org/10.1111/aab.12854</a></p><p>Bila J, Mondjana A, Samils B, Högberg N, Wilson MR &amp; Santos L (2017) First report of ‘<em>Candidatus</em> Phytoplasma palmicola’ detection in the planthopper <em>Diostrombus mkurangai</em> in Mozambique. <em>Bulletin of Insectology</em> <strong>70</strong>(1), 45-48.</p><p>Brown SE, Been BO &amp; McLaughlin WA (2006) Detection and variability of the lethal yellowing group (16Sr IV) phytoplasmas in the <em>Cedusa</em> sp. (Hemiptera: Auchenorrhyncha: Derbidae) in Jamaica. <em>Annals of Applied Biology</em> <strong>149</strong>(1), 53-62.</p><p>Christensen NM, Nicolaisen M, Hansen M &amp; Schulz A (2004) Distribution of phytoplasmas in infected plants as revealed by real-time PCR and bioimaging. <em>Molecular Plant-Microbe Interactions</em> <strong>17</strong>(11), 1175-1184.</p><p>Córdova I, Oropeza C, Puch-Hau C, Harrison N, Collí-Rodríguez A, Narvaez M, Nic-Matos G, Reyes C &amp; Sáenz L (2014) A real-time PCR assay for detection of coconut lethal yellowing phytoplasmas of group 16SrIV subgroups A, D and E found in the Americas. <em>Journal of Plant Pathology</em> <strong>96</strong>(2), 343-352.</p><p>Dzido JL, Sánchez R, Dollet M, Julia JF, Narvaez M, Fabre S &amp; Oropeza C (2020) <em>Haplaxius crudus</em> (Hemiptera: Cixiidae) transmits the lethal yellowing phytoplasmas, 16SrIV, to <em>Pritchardia pacifica</em> Seem. &amp; H. Wendl (Arecaceae) in Yucatan, Mexico. <em>Neotropical Entomology</em> <strong>49</strong>, 795-805.</p><p>Dickinson M (2015) Loop-mediated isothermal amplification (LAMP) for detection of phytoplasmas in the field. In: Lacomme C (ed.). Plant Pathology: Techniques and Protocols, Vol. 1302, Humana Press, New York, NY, USA. pp. 99–112.</p><p>EFSA (2017) Scientific Opinion on pest categorisation of Palm lethal yellowing phytoplasmas. <em>EFSA Journal&nbsp;</em><strong>15</strong>(10), 5028, 27 pp. <a href=\"https://doi.org/10.2903/j.efsa.2017.5028\" rel=\"noopener noreferrer\" target=\"_blank\">https://doi.org/10.2903/j.efsa.2017.5028</a></p><p>EPPO (2024) <em>Haplaxius crudus</em>. EPPO datasheets on pests recommended for regulation. <a href=\"https://gd.eppo.int\">https://gd.eppo.int</a> (accessed 2024-02-06).</p><p>Fernández-Barrera M, Córdova-Lara I, Chan-Rodríguez JL, Castillo-Vera A, Blanco-Rodríguez E, Nic-Matos G, Oropeza-Salín C &amp; Sáenz-Carbonell L (2022) Detection of 16SrIV-A phytoplasma DNA in <em>Colpoptera</em> sp. (Hemiptera: Nogodinidae) insects in Yucatan Peninsula, Mexico. <em>Brazilian Journal of Biology&nbsp;</em><strong>84</strong>, Article e257470. <a href=\"https://doi.org/10.1590/1519-6984.25\" rel=\"noopener noreferrer\" target=\"_blank\">https://doi.org/10.1590/1519-6984.25</a></p><p>Gurr GM, Johnson AC, Ash GJ, Wilson BA, Ero MM, Pilotti CA, Dewhurst CF &amp; You MS (2016) Coconut lethal yellowing diseases: a phytoplasma threat to palms of global economic and social significance. <em>Frontiers in plant science</em> <strong>7</strong>, 1521. <a href=\"https://doi.org/10.3389/fpls.2016.01521\" rel=\"noopener noreferrer\" target=\"_blank\">https://doi.org/10.3389/fpls.2016.01521</a></p><p>Harrison NA, Womack M &amp; Carpio ML (2002) Detection and characterization of a lethal yellowing (16SrIV) group phytoplasma in Canary Island date palms affected by lethal decline in Texas. <em>Plant Disease&nbsp;</em><strong>86</strong>(6), 676-681.</p><p>Hodgetts J, Boonham N, Mumford R &amp; Dickinson M (2009) Panel of 23S rRNA gene-based real-time PCR assays for improved universal and group-specific detection of phytoplasmas. <em>Applied and Environmental Microbiology</em> <strong>75</strong>(9), 2945-2950.</p><p>Howard FW (1980) Population densities of <em>Myndus crudus</em> Van Duzee (Homoptera: Cixiidae) in relation to coconut lethal yellowing distribution in Florida. <em>Principes&nbsp;</em><strong>24</strong>, 174-178.</p><p>Howard FW, Norris R &amp; Thomas D (1983) Evidence of transmission of palm lethal yellowing agent by a planthopper, <em>Myndus crudus</em> (Homoptera, Cixiidae). <em>Tropical Agriculture</em> <strong>60</strong>, 168–171.</p><p>Howard FW (1983) World distribution and possible geographical origin of palm lethal yellowing disease and its vectors. <em>FAO Plant Protection Bulletin</em> <strong>31</strong>, 101-113.</p><p>Howard FW (1990) Evaluation of grasses for cultural control of <em>Myndus crudus,</em> a vector of lethal yellowing of palms. <em>Entomologia Experimentalis et Applicata</em> <strong>56</strong>, 131-137.</p><p>Howard FW, Thomas DL, Donselman HM &amp; Collins ME (1979) Susceptibilities of palm species to mycoplasma-like organism-associated diseases in Florida. <em>FAO Plant Protection Bulletin</em> <strong>27</strong>, 109-117.</p><p>Jones LM, Pease B, Perkiins SL, Constable FE, Kinoti WM, Warmington D, Allgood B, Powell S, Taylor P, Pearce C &amp; Davis RI (2021) ‘<em>Candidatus</em> Phytoplasma dypsidis’, a novel taxon associated with a lethal wilt disease of palms in Australia. <em>International Journal of Systematic and Evolutionary Microbiology</em> <strong>71</strong>(5). <a href=\"https://doi.org/10.1099/ijsem.0.004818\" rel=\"noopener noreferrer\" target=\"_blank\">https://doi.org/10.1099/ijsem.0.004818</a></p><p>Lu H, Wilson BA, Ash GJ, Woruba SB, Fletcher MJ, You M, Yang G &amp; Gurr GM (2016) Determining putative vectors of the Bogia Coconut Syndrome phytoplasma using loop-mediated isothermal amplification of single-insect feeding media. <em>Scientific Reports&nbsp;</em><strong>6</strong>(1), 35801.</p><p>McCoy RE (1975) Effect of oxytetracycline dose and stage of development on remission of lethal yellowing in coconut palm. <em>Plant Disease Reporter</em> <strong>59</strong>, 717-720.</p><p>Miyazaki A, Shigaki T, Koinuma H, Iwabuchi N, Rauka GB, Kembu A, Saul J, Watanabe K, Nijo T, Maejima K &amp; Yamaji Y (2018) ‘<em>Candidatus&nbsp;</em>Phytoplasma noviguineense’, a novel taxon associated with Bogia coconut syndrome and banana wilt disease on the island of New Guinea. <em>International Journal of Systematic and Evolutionary Microbiology</em> <strong>68</strong>(1), 170-175.</p><p>Mou DF, Di Lella B, Halbert SE, Bextine B, Helmick EE &amp; Bahder BW (2022a) Acquisition and transmission of the lethal bronzing phytoplasma by <em>Haplaxius crudus</em> using infected palm spear leaves and artificial feeding media<em>. Phytopathology</em> <strong>112</strong>(10), 2052-2061.</p><p>Mou DF, Helmick EE &amp; Bahder BW (2022b) Multilocus sequence analysis reveals new hosts of palm lethal decline phytoplasmas in Florida, USA. <em>Plant Health Progress</em> <strong>23</strong>(4), 399-408.</p><p>Mpunami AA, Tymon A &amp; Dickinson MJ (1999) Genetic diversity in the coconut lethal yellowing disease phytoplasmas of East Africa. <em>Plant Pathology</em> <strong>48</strong>(1), 109-114.</p><p>Mpunami A, Tymon A, Jones P &amp; Dickinson MJ (2000) Identification of potential vectors of the coconut lethal disease phytoplasma. <em>Plant Pathology</em> <strong>49</strong>(3), 355-361.</p><p>Ogle L &amp; Harries H (2005) Introducing the vector: how coconut lethal yellowing disease may have reached the Caribbean. <em>Ethnobotany Research and Applications</em> <strong>3</strong>, 139–142</p><p>Oropeza C, Cordova I, Puch‐Hau C, Castillo R, Chan JL &amp; Sáenz L (2017) Detection of lethal yellowing phytoplasma in coconut plantlets obtained through in vitro germination of zygotic embryos from the seeds of infected palms. <em>Annals of Applied Biology</em> <strong>171</strong>(1), 28-36.</p><p>Palma-Cancino PJ, Ramos-Hernández E &amp; Ortiz-García CF (2023) History of lethal yellowing with emphasis on the susceptibility of royal palms (<em>Roystonea</em> spp.). <em>Agro Productividad.</em> <a href=\"https://doi.org/10.32854/agrop.v16i4.2488\" rel=\"noopener noreferrer\" target=\"_blank\">https://doi.org/10.32854/agrop.v16i4.2488</a></p><p>Pilet F (2021) Diagnostic for Phytoplasmae associated with coconut lethal yellowing type syndromes from Africa, and elsewhere. Cirad Webinar (2021-01-21-21).</p><p>Plavsic-Banjac B, Hunt P &amp; Maramorosch K (1972) Mycoplasma-like bodies associated with lethal yellowing disease of coconut palms. <em>Phytopathology&nbsp;</em><strong>62</strong>, 298-299.</p><p>Purcell AH (1985) The ecology of plant diseases spread by leafhoppers and planthoppers. In: <em>The leafhoppers and planthoppers</em> (Ed. by Nault, L.R.; Rodriguez, J.G.), pp. 351-380. Wiley, New York, USA.</p><p>Ramos Hernández E, Lesher Gordillo MJ, Ortiz García CF, Oropeza Salín C, Sánchez Soto S, Magaña Alejandro MÁ &amp; Narváez MD (2020) Cixiidae and Derbidae (Hemiptera) putative vectors of phytoplasma of group 16 SrIV. <em>Revista Colombiana De Entomología</em> <strong>46</strong>(2), e7065. <a href=\"https://doi.org/10.25100/socolen.v46i2.7065\" rel=\"noopener noreferrer\" target=\"_blank\">https://doi.org/10.25100/socolen.v46i2.7065</a></p><p>Soto N, Helmick EE, Harrison NA &amp; Bahder BW (2021) Genetic variability of palm lethal decline phytoplasmas in the Caribbean basin and Florida, USA, based on a multilocus analysis. <em>Phytopathology </em><strong>111</strong>(12), 2203-2212. <a href=\"https://doi.org/10.1094/PHYTO-04-21-0130-R\" rel=\"noopener noreferrer\" target=\"_blank\">https://doi.org/10.1094/PHYTO-04-21-0130-R</a></p><p>Tropicsafe (2021a) Rapid infield detection of coconut phytoplasma in Africa. Final Handbook with innovation Factsheets. <a href=\"http://www.tropicsafe.eu/wp-content/uploads/2022/02/RAPID-IN-FIELD-DETECTION-OF-COCO-PHYTO-AF.pdf\" rel=\"noopener noreferrer\" target=\"_blank\">http://www.tropicsafe.eu/wp-content/uploads/2022/02/RAPID-IN-FIELD-DETECTION-OF-COCO-PHYTO-AF.pdf</a></p><p>Tropicsafe (2021b) Coconut sector: market analysis and socioeconomic aspects -A general framework of the coconut sector in Jamaica, Ghana, and Mexico. Final Handbook with innovation Factsheets. <a href=\"http://www.tropicsafe.eu/wp-content/uploads/2022/02/COCONUT-SECTOR-MARKET.pdf\" rel=\"noopener noreferrer\" target=\"_blank\">http://www.tropicsafe.eu/wp-content/uploads/2022/02/COCONUT-SECTOR-MARKET.pdf</a></p><p>Tropicsafe (2021c) Coconut lethal yellowing resistance screening and disease management - The use of lethal yellowing resistant germplasm. Final Handbook with innovation Factsheets. <a href=\"https://www.tropicsafe.eu/wp-content/uploads/2022/02/COCONUT-LETHAL-YELLOWING-RESISTANCE.pdf\" rel=\"noopener noreferrer\" target=\"_blank\">https://www.tropicsafe.eu/wp-content/uploads/2022/02/COCONUT-LETHAL-YELLOWING-RESISTANCE.pdf</a></p><p>Wei W, Davis RE, Lee IM &amp; Zhao Y (2007) Computer-simulated RFLP analysis of 16S rRNA genes: identification of ten new phytoplasma groups. <em>International Journal of Systematic and Evolutionary Microbiology&nbsp;</em><strong>57</strong>(8), 1855-1867. <a href=\"https://doi.org/10.1099/ijs.0.65000-0\" rel=\"noopener noreferrer\" target=\"_blank\">https://doi.org/10.1099/ijs.0.65000-0</a></p><p>Yu SS, Pan YW, Zhu H &amp; Song WW (2023) Universal, rapid, and visual detection methods for phytoplasmas associated with coconut lethal yellowing diseases targeting 16S rRNA gene sequences. <em>Plant Disease</em> <strong>107</strong>(2), 276-280.</p></div>\n",
    "<!-- -->\n",
    "<a class=\"btntitle\" href=\"#\"><h3><i class=\"fa fa-plus-circle\" aria-hidden=\"true\"></i> ACKNOWLEDGEMENTS<small class=\"pull-right\" title=\"Last modification\"><i class=\"fa fa-clock-o\" aria-hidden=\"true\"></i> 2024-06-18</small></h3></a>\n",
    "<div class=\"para1\">\n",
    "<p>This datasheet was extensively revised in 2024 by the EPPO Secretariat with the help of Fabian Pilet (CIRAD). His valuable contribution is gratefully acknowledged.</p></div>\n",
    "<!-- -->\n",
    "<a class=\"btntitle\" href=\"#\"><h3><i class=\"fa fa-plus-circle\" aria-hidden=\"true\"></i> How to cite this datasheet?</h3></a>\n",
    "<div class=\"para1\">EPPO (2024) <i>Palm lethal yellowing type syndromes</i>. EPPO datasheets on pests recommended for regulation. <a href=\"https://gd.eppo.int\" target=\"_blank\">https://gd.eppo.int</a> (accessed 2024-11-25)\n",
    "</div>\n",
    "\n",
    "<!-- -->\n",
    "<a class=\"btntitle\" href=\"#\"><h3><i class=\"fa fa-plus-circle\" aria-hidden=\"true\"></i> Datasheet history<small class=\"pull-right\" title=\"Last modification\"><i class=\"fa fa-clock-o\" aria-hidden=\"true\"></i> 2024-06-18</small></h3></a>\n",
    "<div class=\"para1\">\n",
    "<p style=\"text-align: justify;\">This datasheet was first published in the EPPO Bulletin in 1986 and revised in the two editions of 'Quarantine Pests for Europe' in 1992 and 1997. It is now maintained in an electronic format in the EPPO Global Database. The sections on 'Identity', ‘Hosts’, and 'Geographical distribution' are automatically updated from the database. For other sections, the date of last revision is indicated on the right.</p><p>CABI/EPPO (1992/1997) Quarantine Pests for Europe (1st and 2nd edition). CABI, Wallingford (GB).</p><p>EPPO (1986) Data sheets on quarantine organisms No. 159,<em>&nbsp;Palm lethal yellowing mycoplasm. EPPO Bulletin</em> <strong>16</strong>, 61-66.&nbsp;<a href=\"https://onlinelibrary.wiley.com/doi/epdf/10.1111/j.1365-2338.1986.tb01138.x\" rel=\"noopener noreferrer\" target=\"_blank\">https://onlinelibrary.wiley.com/doi/epdf/10.1111/j.1365-2338.1986.tb01138.x</a></p></div>\n",
    "<!-- -->\n",
    "    <img src=\"/media/img/eu_funding_250.png\">\n",
    "\n",
    "\"\"\"\n",
    "ignore_list = ['references', 'acknowledgements', 'how to cite this datasheet?', 'datasheet history']\n",
    "from tqdm import tqdm\n",
    "def get_datasheet(url):\n",
    "    import requests\n",
    "    from bs4 import BeautifulSoup\n",
    "    response = requests.get(url)\n",
    "    soup = BeautifulSoup(response.content, 'html.parser')\n",
    "    datasheet = soup.find('div', class_='datasheet')\n",
    "    if datasheet is None:\n",
    "        return None\n",
    "    datasheet_dict = {}\n",
    "    for btn in datasheet.find_all('h3'):\n",
    "        # remove 'pull-right' class from h3 children\n",
    "        for child in btn.find_all('small'):\n",
    "            child.decompose()        \n",
    "        key = btn.text.lower()\n",
    "        if key in ignore_list:\n",
    "            continue\n",
    "        datasheet_dict[key] = {}\n",
    "        if btn.find_next('div', class_='para0'):\n",
    "            datasheet_dict[key]['description'] = btn.find_next('div', class_='para0').text.replace('\\n', ' ')\n",
    "        elif btn.find_next('div', class_='para1'):\n",
    "            if btn.find_next('div', class_='para1').find_next().name == 'p':\n",
    "                datasheet_dict[key]['description'] = btn.find_next('div', class_='para1').text.replace('\\n', ' ')\n",
    "            elif btn.find_next('div', class_='para1').find_next().name == 'h4':\n",
    "                for h4 in btn.find_next('div', class_='para1').find_all('h4'):\n",
    "                    key1 = h4.text.lower()\n",
    "                    datasheet_dict[key][key1] = h4.find_next('p').text.replace('\\n', ' ')      \n",
    "        \n",
    "    return datasheet_dict\n",
    "\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 354/354 [01:33<00:00,  3.77it/s]\n"
     ]
    }
   ],
   "source": [
    "datasheet_dict = {}\n",
    "for url in tqdm(dfs.Datasheet):\n",
    "    key = url.split('/')[-2]\n",
    "    datasheet_dict[key] = get_datasheet(url)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 186,
   "metadata": {},
   "outputs": [],
   "source": [
    "# new dict copy with keys without \\u00a0\n",
    "import unicodedata\n",
    "def remove_non_breaking_space(cist):\n",
    "    new_cist = {}\n",
    "    for key, value in cist.items():\n",
    "        if isinstance(value, dict):\n",
    "            new_cist[unicodedata.normalize('NFKD', key).encode('ascii', 'ignore').decode('ascii')] = remove_non_breaking_space(value)\n",
    "        elif isinstance(value, str):\n",
    "            new_cist[unicodedata.normalize('NFKD', key).encode('ascii', 'ignore').decode('ascii')] = unicodedata.normalize('NFKD', value).encode('ascii', 'ignore').decode('ascii').replace('\\\"', \"'\").replace(\"\\r\",' ')\n",
    "    return new_cist\n",
    "\n",
    "datasheet_dict = remove_non_breaking_space(datasheet_dict)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 187,
   "metadata": {},
   "outputs": [],
   "source": [
    "# write file\n",
    "import json\n",
    "with open('/workdir/eppo_datasheet.json', 'w') as f:\n",
    "    json.dump(datasheet_dict, f, indent=4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 165,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{\n",
      "    \"TSWV00\": {\n",
      "        \"identity\": {\n",
      "            \"description\": \" Preferred name: Orthotospovirus tomatomaculae Taxonomic position: Viruses and viroids: Riboviria: Orthornavirae: Negarnaviricota: Polyploviricotina: Bunyaviricetes: Elliovirales: Tospoviridae: Orthotospovirus Other scientific names:  TSWV, Tomato spotted wilt orthotospovirus, Tomato spotted wilt tospovirus, Tomato spotted wilt virus Common names in English:  bronze leaf of tomato, kromnek virus, spotted wilt of tomato, yellow spot of pineapple view more common names online...  Notes on taxonomy and nomenclature  Tomato spotted wilt disease was first described in Australia in 1915 and was later identified as a viral disease caused by tomato spotted wilt virus (TSWV)\\u00a0(Samuel et al., 1930). In 1990, TSWV was assigned to the genus Tospovirus,\\u00a0which was later renamed\\u00a0Orthotospovirus (family Tospoviridae, order Bunyavirales; ICTV online: https://talk.ictvonline.org/taxonomy/; Abudurexiti et al., 2019). Species demarcation within the genus is based on nucleoprotein (N) sequence (new species are defined as having less than 90% amino acid sequence similarity to all other described species within the genus); in addition, species are often biologically distinguished by their host range and vector specificity (Plyusnin et al., 2011; Kormelink et al., 2021).  EPPO Categorization: A2 list EU Categorization: RNQP (Annex IV) view more categorizations online... EPPO Code: TSWV00\"\n",
      "        },\n",
      "        \"hosts\": {\n",
      "            \"description\": \" TSWV has one of the largest known host ranges of any plant virus, with more than 1000 plant species recorded as hosts. The host range includes dicotyledons and monocotyledons, crop plants, ornamentals and weeds. Most of the plant species susceptible to TSWV belong to the Asteraceae\\u00a0and Solanaceae families (Parrella et al., 2003).In the EPPO region, the main crop hosts are artichokes (Cynara scolymus), eggplants (Solanum melongena), peppers (Capsicum annuum), chicory (Cichorium spp.), cucurbits (Cucurbitaceae), faba beans (Vicia faba), lettuce (Lactuca sativa), potatoes (Solanum tuberosum), tobacco (Nicotiana tabacum), tomatoes (Solanum lycopersicum); the main ornamental hosts are Anemone, Aster, Begonia hybrids, Calceolaria, Callistephus, chrysanthemum (Dendranthema x grandiflorum), Cyclamen, Dahlia, Gerbera, Pelargonium, Pericallis cruenta, Ranunculus, Sinningia spp., Symphyotrichum, Tagetes patula and Zinnia. Wild species such as Senecio vulgaris, Solanum nigrum, Sonchus spp. and Stellaria media can be important reservoirs for TSWV.Host list: Acalypha australis, Acanthospermum hispidum, Acanthus mollis, Achimenes sp., Aconitum carmichaelii, Adenium obesum, Aegopodium podagraria, Aeschynanthus pulcher, Aeschynanthus speciosus, Agapanthus praecox subsp. orientalis, Agapanthus praecox, Agastache foeniculum, Ageratum houstonianum, Aglaonema commutatum, Agrostemma githago, Ajania pacifica, Alcea rosea, Alkekengi officinarum, Alliaria petiolata, Allium ascalonicum, Allium cepa, Allium porrum, Allium sativum, Alstroemeria aurea, Alstroemeria hybrids, Alternanthera sessilis, Althaea officinalis, Amaranthus albus, Amaranthus blitum, Amaranthus caudatus, Amaranthus cruentus, Amaranthus graecizans, Amaranthus hybridus, Amaranthus palmeri, Amaranthus powellii, Amaranthus retroflexus, Amaranthus spinosus, Amaranthus thunbergii, Amaranthus viridis, Amaryllis belladonna, Ambrosia artemisiifolia, Ambrosia psilostachya, Ambrosia trifida, Ammi majus, Ananas comosus, Anchusa azurea, Anchusa capensis, Anchusa officinalis, Anchusa undulata, Anemone coronaria, Antennaria neglecta, Anthemis arvensis, Anthurium andraeanum, Anthurium hookeri, Anthurium scherzerianum, Antirrhinum majus, Aphelandra squarrosa, Apium graveolens var. dulce, Apium graveolens, Aquilegia vulgaris, Arabis sp., Arachis batizocoi, Arachis duranensis, Arachis hypogaea, Arachis monticola, Arachis pusilla, Arachis stenosperma, Arachis villosulicarpa, Arctium lappa, Arctium minus, Arctotheca calendula, Arctotis x hybrida, Ardisia sp., Argyranthemum frutescens, Aristolochia clematitis, Aristolochia elegans, Artemisia absinthium, Artemisia dracunculus, Artemisia princeps, Arum maculatum, Arum palaestinum, Asarum canadense, Asclepias curassavica, Asparagus setaceus, Asplenium nidus, Aster amellus, Aubrieta deltoidea, Avena fatua, Ballota nigra, Barbarea vulgaris, Begonia gracilis, Begonia semperflorens hybrids, Begonia tuberhybrida hybrids, Begonia x hiemalis, Bellis caerulescens, Berberis sp., Beta vulgaris, Betonica officinalis, Bidens discoidea, Bidens pilosa, Bidens subalternans, Bidens vulgata, Blitum capitatum, Blitum virgatum, Bothriocline longipes, Bouvardia sp., Brachyscome iberidifolia, Brassica juncea, Brassica napus, Brassica oleracea var. botrytis, Brassica oleracea, Brassica perviridis, Brassica rapa subsp. chinensis, Brassica rapa subsp. pekinensis, Brassica rapa subsp. sylvestris, Browallia americana, Browallia speciosa, Brugmansia arborea, Brugmansia aurea, Brugmansia suaveolens, Buddleia davidii, Cajanus cajan, Calceolaria crenatiflora, Calceolaria herbeohybrida hybrids, Calectasia cyanea, Calendula arvensis, Calendula officinalis, Calla palustris, Callistephus chinensis, Calotropis gigantea, Calystegia sepium, Campanula americana, Campanula glomerata, Campanula grandis subsp. grandis, Campanula isophylla, Campanula medium, Campanula persicifolia, Campanula pyramidalis, Campanula rapunculoides, Canavalia ensiformis, Canavalia gladiata, Canna sp., Capsella bursa-pastoris, Capsicum annuum, Capsicum chinense, Capsicum frutescens, Cardamine flexuosa, Cardamine hirsuta, Cardamine oligosperma, Cardamine parviflora, Carduus acanthoides, Carduus nutans, Carica papaya, Carthamus tinctorius, Carum carvi, Catharanthus roseus, Celosia argentea, Centaurea cineraria, Centaurea cyanus, Centella asiatica, Centranthus ruber, Cerastium glomeratum, Cerastium holosteoides, Cestrum aurantiacum, Cestrum elegans, Cestrum nocturnum, Cestrum parqui, Chaerophyllum temulum, Chamaedorea costaricana, Chamaedorea elegans, Chelone sp., Chenopodiastrum murale, Chenopodiastrum simplex, Chenopodium album, Chenopodium ficifolium, Chenopodium giganteum, Chenopodium vulvaria, Chondrilla juncea, Chrysanthemum indicum, Chrysanthemum x morifolium, Cicer arietinum, Cichorium endivia var. latifolia, Cichorium endivia, Cichorium intybus, Cineraria sp., Cirsium arvense, Cirsium japonicum, Cirsium palustre, Cirsium setidens, Cirsium vulgare, Citrullus lanatus, Clarkia amoena subsp. lindleyi, Claytonia perfoliata, Clematis flammula, Clematis vitalba, Clerodendrum thomsoniae, Clivia sp., Codonanthe crassifolia, Codonopsis pilosula, Coffea arabica, Coleus australis, Coleus scutellarioides, Columnea hirta, Commelina communis, Commelina cyanea, Conium maculatum, Convolvulus althaeoides, Convolvulus arvensis, Convolvulus cantabrica, Coprosma repens, Cordyline fruticosa, Coreopsis basalis, Coreopsis lanceolata, Coreopsis tinctoria, Coriandrum sativum, Cornus sanguinea, Cosmos bipinnatus, Cota tinctoria, Crepis capillaris, Crepis divaricata, Crepis foetida subsp. rhoeadifolia, Crepis occidentalis subsp. pumila, Crepis pulchra, Crinum jagus, Crinum moorei, Crotalaria incana, Crotalaria juncea, Crotalaria pallida var. obovata, Crotalaria spectabilis, Croton glandulosus, Cucumis melo, Cucumis sativus, Cucurbita maxima, Cucurbita moschata, Cucurbita pepo, Cyanus montanus, Cyclamen persicum, Cymbidium sp., Cynanchum rostellatum, Cynara cardunculus, Cynara scolymus, Cynodon dactylon, Cynoglossum sp., Cyperus esculentus, Cyperus rotundus, Cyrtanthus sp., Cytisus scoparius, Dahlia hybrids, Dahlia pinnata, Datura ferox, Datura innoxia, Datura metel, Datura stramonium var. tatula, Datura stramonium, Datura wrightii, Delphinium hybrids, Dendrobium sp., Desmodium tortuosum, Desmodium uncinatum, Dianthus chinensis, Diascia barberae, Dieffenbachia sp., Digitalis sp., Digitaria sanguinalis, Dimorphotheca ecklonis, Dimorphotheca fruticosa, Dimorphotheca pluvialis, Dimorphotheca sinuata, Diplotaxis erucoides, Diplotaxis muralis, Dorotheanthus tricolor, Dracaena draco, Dracaena fragrans, Dracaena marginata, Duboisia leichhardtii, Duboisia myoporoides, Dysphania ambrosioides, Echinacea purpurea, Echinochloa colonum, Echinocystis lobata, Eclipta prostrata, Eleusine indica, Emilia coccinea, Emilia sonchifolia, Epilobium hirsutum, Epipremnum pinnatum, Erigeron bonariensis, Erigeron canadensis, Erigeron strigosus, Erodium ciconium, Erodium moschatum, Erysimum cheiranthoides, Erysimum x cheiri, Eupatorium capillifolium, Eupatorium maculatum, Euphorbia heterophylla, Euphorbia peplus, Euphorbia pulcherrima, Euphorbia tithymaloides, Euryops sp., Eustoma russellianum, Exacum affine, Fagopyrum esculentum, Fallopia convolvulus, Farfugium japonicum, Fatshedera lizei, Fatsia japonica, Felicia amelloides, Felicia bergeriana, Felicia fruticosa, Ficus benjamina, Ficus cyathistipula, Ficus elastica, Ficus lyrata, Flaveria bidentis, Forsythia koreana, Fragaria vesca, Freesia hybrids, Fuchsia hybrids, Fumaria muralis, Fumaria officinalis, Gaillardia aristata, Gaillardia x grandiflora, Galinsoga parviflora, Galinsoga quadriradiata, Galium aparine, Galium lucidum, Galium parisiense, Galium spurium, Galium tricornutum, Galium verum, Gamochaeta falcata, Gamochaeta pensylvanica, Gamochaeta purpurea, Gardenia jasminoides, Gazania rigens, Geranium carolinianum, Geranium robertianum, Geranium rotundifolium, Gerbera jamesonii, Gerbera x hybrida, Gesneria sp., Gladiolus grandiflorus, Glaucium flavum, Glebionis coronaria, Gleditsia triacanthos, Gloriosa superba, Gloxinia sp., Glycine max, Glycine soja, Gnaphalium uliginosum, Gomphrena globosa, Goniolimon tataricum, Gossypium hirsutum, Guzmania sp., Gymnema sylvestre, Gynura aurantiaca, Gypsophila elegans, Gypsophila paniculata, Helianthus annuus, Helichrysum sp., Heliotropium europaeum, Helminthotheca echioides, Hemistepta lyrata, Heptapleurum actinophyllum, Heptapleurum arboricola, Hesperis matronalis, Hibiscus trionum, Hippeastrum aulicum, Hippeastrum hybrids, Hippeastrum reginae, Hippeastrum striatum, Hirschfeldia incana, Holcus lanatus, Hoya australis, Hoya bella, Hoya carnosa, Hoya linearis, Humulus scandens, Hydrangea macrophylla, Hymenocallis occidentalis, Hyoscyamus niger, Hypericum triquetrifolium, Iberis semperflorens, Impatiens New Guinea hybrids, Impatiens balsamina, Impatiens capensis, Impatiens hybrids, Impatiens walleriana, Inula helenium, Ipomoea batatas, Ipomoea indica, Ipomoea lacunosa, Ipomoea purpurea, Iris domestica, Iris ensata, Iris x hollandica, Jacobaea maritima, Jacobaea vulgaris, Jacquemontia tamnifolia, Jasminum odoratissimum, Justicia brandegeana, Kalanchoe blossfeldiana, Kalanchoe daigremontiana, Kalanchoe synsepala, Kalanchoe thyrsiflora, Lablab purpureus, Lactuca canadensis, Lactuca floridana, Lactuca indica, Lactuca saligna, Lactuca sativa var. capitata, Lactuca sativa var. crispa, Lactuca sativa var. longifolia, Lactuca sativa, Lactuca serriola, Lagascea mollis, Lagenaria siceraria, Lamium amplexicaule, Lamium purpureum, Lantana camara, Lathyrus odoratus, Lathyrus sativus, Lavandula angustifolia subsp. pyrenaica, Layia elegans, Leonotis nepetifolia, Lepidium didymum, Lepidium draba, Lepidium sativum, Lepidium virginicum, Leptochiton quitoensis, Leucanthemum maximum, Leucanthemum vulgare, Leucanthemum x superbum, Ligustrum vulgare, Lilium lancifolium, Lilium longiflorum, Limonium perezii, Limonium platyphyllum, Limonium sinuatum, Limonium vulgare, Lisianthius sp., Lobelia dortmanna, Lobelia erinus, Lobelia valida, Lobularia maritima, Lolium perenne, Luffa aegyptiaca, Lupinus albus, Lupinus angustifolius, Lupinus leucophyllus, Lupinus polyphyllus, Lupinus subcarnosus, Lychnis chalcedonica, Lycianthes rantonnetii, Lycium chinense, Lycium ferocissimum, Lycopus europaeus, Lysimachia arvensis, Lysimachia congestiflora, Lysimachia loeflingii, Lysimachia nummularia, Macrotyloma uniflorum, Maianthemum canadense, Malcolmia maritima, Malva neglecta, Malva nicaeensis, Malva parviflora, Malva pusilla, Malva sylvestris, Malva verticillata, Maranta leuconeura, Marrubium vulgare, Martynia annua, Matricaria chamomilla, Matricaria discoidea, Matthiola incana, Mazus pumilus, Medicago lupulina, Medicago polymorpha, Medinilla magnifica, Melampodium sp., Melilotus indicus, Melilotus officinalis, Melissa officinalis, Mentha arvensis, Mentha microphylla, Mentha spicata, Mentha suaveolens, Mentha x piperita, Mercurialis annua, Mimulus sp., Mirabilis jalapa, Mollugo verticillata, Moluccella laevis, Monarda didyma, Monarda fistulosa, Montia sp., Morus alba, Myosotis alpestris, Myosoton aquaticum, Nasturtium officinale, Nepeta cataria, Nepeta nuda, Nerine bowdenii, Nerium oleander, Nicandra physalodes, Nicotiana acuminata, Nicotiana alata, Nicotiana benthamiana, Nicotiana bonariensis, Nicotiana debneyi, Nicotiana excelsior, Nicotiana exigua, Nicotiana glauca, Nicotiana goodspeedii, Nicotiana langsdorffii, Nicotiana longiflora, Nicotiana maritima, Nicotiana megalosiphon, Nicotiana nudicaulis, Nicotiana paniculata, Nicotiana pauciflora, Nicotiana plumbaginifolia, Nicotiana quadrivalvis, Nicotiana solanifolia, Nicotiana suaveolens, Nicotiana sylvestris, Nicotiana tabacum, Nicotiana tomentosiformis, Nicotiana undulata, Nicotiana velutina, Nicotiana wigandioides, Nicotiana x edwardsonii, Nicotiana x sanderi, Nolana sp., Notobasis syriaca, Nuttallanthus canadensis, Ocimum basilicum, Oenanthe javanica, Oenothera biennis, Oenothera laciniata, Oenothera sp., Oncidium sp., Onopordum acanthium, Onopordum illyricum, Opuntia sp., Orlaya sp., Ornithogalum thyrsoides, Orobanche sp., Osteospermum sp., Oxalis acetosella, Oxalis corniculata, Oxalis latifolia, Oxalis stricta, Oxalis tuberosa, Oxybasis glauca, Oxybasis urbica, Oxypetalum sp., Pachypodium lamerei, Paederia foetida, Paeonia lactiflora, Papaver dubium, Papaver nudicaule, Papaver orientale, Papaver rhoeas, Papaver somniferum, Parietaria officinalis, Pelargonium crispum, Pelargonium grandiflorum, Pelargonium peltatum, Pelargonium x hortorum, Pelargonium zonale, Penstemon hartwegii, Penstemon hirsutus, Peperomia fraseri, Peperomia obtusifolia, Peperomia rotundifolia, Pericallis cruenta, Pericallis x hybrida, Persicaria hydropiper, Persicaria lapathifolia, Persicaria longiseta, Persicaria maculosa, Persicaria pensylvanica, Petasites japonicus, Petroselinum crispum, Petunia hybrids, Petunia integrifolia, Phalaenopsis sp., Phaseolus lunatus, Phaseolus vulgaris, Philodendron sagittifolium, Phlox drummondii, Physalis acutifolia, Physalis alkekengi var. franchetii, Physalis angulata, Physalis heterophylla, Physalis ixocarpa, Physalis minima, Physalis peruviana, Physalis pruinosa, Physalis pubescens, Physostegia virginiana, Phytolacca americana, Pilea pumila, Pinus sp., Pisum sativum subsp. arvense, Pisum sativum, Pittosporum tobira, Plantago lanceolata, Plantago major, Plantago rugelii, Platycerium superbum, Platycodon grandiflorus, Plumeria rubra f. acutifolia, Poa annua, Polygonum aviculare, Portulaca grandiflora, Portulaca oleracea, Portulaca pilosa, Potentilla reptans, Primula malacoides, Primula obconica, Primula sinensis, Primula vulgaris, Prunella vulgaris, Pyrrhopappus carolinianus, Ranunculus abortivus, Ranunculus acris, Ranunculus arvensis, Ranunculus asiaticus, Ranunculus bulbosus, Ranunculus hybridus, Ranunculus muricatus, Ranunculus sardous, Raphanus raphanistrum, Raphanus sativus, Rhamnus sp., Rhaponticum carthamoides, Rhodanthe chlorocephala subsp. rosea, Rhodanthe manglesii, Rhododendron sp., Richardia scabra, Ricinus communis, Robinia pseudoacacia, Rohdea sp., Rorippa indica, Rorippa palustris, Rosa sp., Rubus idaeus, Rubus ulmifolius, Rudbeckia amplexicaulis, Rudbeckia hirta, Rudbeckia nitida, Rudbeckia occidentalis, Rumex conglomeratus, Rumex crispus, Ruscus hypoglossum, Salpiglossis sinuata, Salsola kali subsp. ruthenica, Salvia officinalis, Salvia pratensis, Salvia sclarea, Salvia splendens, Salvia verbenaca, Sambucus nigra, Sanguisorba minor, Saponaria officinalis, Saxifraga stolonifera, Scabiosa atropurpurea, Schizanthus pinnatus, Schlumbergera truncata, Schlumbergera x buckleyi, Scutellaria sp., Sechium edule, Sedum sarmentosum, Seemannia sylvatica, Senecio doria, Senecio vulgaris, Senna obtusifolia, Senna occidentalis, Senna tora, Sesamum indicum, Setaria viridis, Sida rhombifolia, Sida spinosa, Sigesbeckia orientalis, Silene coronaria, Silene firma, Silene latifolia subsp. alba, Silene vulgaris, Sinapis arvensis, Sinningia speciosa, Sisymbrium irio, Smilax sieboldii, Solanum aculeatissimum, Solanum americanum, Solanum aviculare, Solanum betaceum, Solanum capsicoides, Solanum carolinense, Solanum chenopodioides, Solanum chilense, Solanum dulcamara, Solanum grandiflorum, Solanum habrochaites, Solanum hieronymi, Solanum laciniatum, Solanum lycocarpum, Solanum lycopersicum, Solanum mammosum, Solanum marginatum, Solanum melongena, Solanum muricatum, Solanum neorickii, Solanum nigrum, Solanum palinacanthum, Solanum pennellii, Solanum pimpinellifolium, Solanum pseudocapsicum, Solanum quitoense, Solanum robustum, Solanum seaforthianum, Solanum triflorum, Solanum trifolium, Solanum tuberosum, Solanum tweedieanum, Solanum viarum, Solanum villosum subsp. miniatum, Solanum violaceum, Solidago altissima, Solidago canadensis, Sonchus asper, Sonchus oleraceus, Sonchus tenerrimus, Sorbus aucuparia, Spathiphyllum floribundum, Spergula arvensis, Spinacia oleracea, Stachys arvensis, Stapelia sp., Stellaria media, Stellaria uliginosa, Stephanotis floribunda, Stevia rebaudiana, Streptocarpus ionanthus, Streptosolen jamesonii, Symphoricarpos albus, Symphyotrichum cordifolium, Symphyotrichum ericoides, Symphyotrichum lateriflorum, Symphyotrichum novae-angliae, Symphytum tuberosum, Syngonium podophyllum, Tagetes erecta, Tagetes minuta, Tagetes patula, Taraxacum officinale, Tephrosia purpurea, Tetragonia tetragonioides, Thymus vulgaris, Tithonia rotundifolia, Tolmiea menziesii, Torilis arvensis, Trachelium caeruleum, Trachymene coerulea, Tradescantia albiflora, Tragopogon dubius, Tragopogon porrifolius, Tragopogon pratensis, Tragopogon x mirus, Tragopogon x miscellus, Trevesia palmata, Tribulus terrestris, Trichosanthes kirilowii, Trifolium incarnatum, Trifolium repens, Trifolium sp., Trifolium subterraneum, Trifolium tembense, Triodanis perfoliata, Tripleurospermum inodorum, Tropaeolum majus, Tulbaghia violacea, Tussilago farfara, Urceolina x grandiflora, Urtica dioica, Valeriana fauriei, Valeriana officinalis, Valerianella locusta, Verbascum blattaria, Verbascum thapsus, Verbena brasiliensis, Verbena hastata, Verbena hybrids, Verbena litoralis, Verbena officinalis, Verbena rigida, Verbesina encelioides, Veronica agrestis, Veronica chamaedrys, Veronica hederifolia, Veronica officinalis, Veronica persica, Vicia amoena, Vicia faba, Vicia hirsuta, Vicia lens, Vigna angularis, Vigna mungo, Vigna radiata, Vigna unguiculata subsp. sesquipedalis, Vigna unguiculata subsp. unguiculata, Vigna unguiculata, Vinca minor, Viola cornuta, Viola sororia, Viola x wittrockiana, Vitis vinifera, Wahlenbergia marginata, Weigela florida, Xanthium orientale subsp. saccharatum, Xanthium spinosum, Xanthium strumarium, Xerochrysum bracteatum, Youngia japonica, Yucca aloifolia, Yucca baccata, Zantedeschia aethiopica, Zantedeschia albomaculata subsp. albomaculata, Zantedeschia albomaculata, Zantedeschia elliottiana, Zantedeschia rehmannii, Zinnia elegans \"\n",
      "        },\n",
      "        \"geographical distribution\": {\n",
      "            \"description\": \" Although tomato spotted wilt disease was reported from Australia more than 100 years ago, it only spread rapidly in the 1980s due to the global spread of Frankliniella occidentalis,\\u00a0one of the major thrips vectors (EFSA, 2012b; Kormelink et al., 2021). It is now present in almost all countries with temperate, tropical and subtropical climates. EPPO Region: Albania, Algeria, Armenia, Austria, Azerbaijan, Belgium, Bosnia and Herzegovina, Bulgaria, Croatia, Cyprus, Czech Republic, Finland, France (mainland), Georgia, Germany, Greece (mainland, Kriti), Guernsey, Hungary, Ireland, Israel, Italy (mainland, Sardegna, Sicilia), Jordan, Lithuania, Malta, Moldova, Montenegro, Netherlands, North Macedonia, Poland, Portugal (mainland, Madeira), Romania, Russia (Far East, Southern Russia), Serbia, Slovenia, Spain (mainland, Islas Baleares, Islas Can\\u00e1rias), Sweden, Switzerland, Tunisia, T\\u00fcrkiye, Ukraine, United Kingdom (Channel Islands, England, Scotland) Africa: Algeria, Burkina Faso, Congo, Democratic republic of the, Cote d'Ivoire, Egypt, Kenya, Libya, Madagascar, Mauritius, Niger, Nigeria, Reunion, Senegal, South Africa, Sudan, Tanzania, Tunisia, Uganda, Zimbabwe Asia: Afghanistan, China (Beijing, Chongqing, Gansu, Guangdong, Guizhou, Heilongjiang, Hubei, Liaoning, Ningxia, Qinghai, Shaanxi, Shandong, Sichuan, Tianjin, Yunnan), India (Andhra Pradesh, Assam, Haryana, Himachal Pradesh, Karnataka, Kerala, Madhya Pradesh, Maharashtra, Tamil Nadu, Telangana, Uttar Pradesh), Indonesia (Java), Iran, Iraq, Israel, Japan (Hokkaido, Honshu, Ryukyu Archipelago), Jordan, Korea, Republic, Lebanon, Malaysia (West), Nepal, Oman, Pakistan, Saudi Arabia, Sri Lanka, Syria, Taiwan, Thailand North America: Canada (Alberta, British Columbia, Manitoba, Nova Scotia, Ontario, Qu\\u00e9bec, Saskatchewan), Mexico, United States of America (Alabama, Arizona, Arkansas, California, Connecticut, Delaware, Florida, Georgia, Hawaii, Idaho, Indiana, Iowa, Kansas, Kentucky, Louisiana, Maine, Massachusetts, Michigan, Minnesota, Mississippi, Missouri, Montana, Nebraska, Nevada, New Hampshire, New Mexico, New York, North Carolina, North Dakota, Ohio, Oklahoma, Oregon, Pennsylvania, South Carolina, South Dakota, Tennessee, Texas, Utah, Vermont, Virginia, Washington, Wisconsin, Wyoming) Central America and Caribbean: Costa Rica, Dominican Republic, Haiti, Jamaica, Puerto Rico South America: Argentina, Bolivia, Brazil (Bahia, Goias, Minas Gerais, Parana, Pernambuco, Sao Paulo), Chile, Colombia, Ecuador, Guyana, Paraguay, Suriname, Uruguay, Venezuela Oceania: Australia (New South Wales, Northern Territory, Queensland, South Australia, Tasmania, Victoria, Western Australia), Cook Islands, New Zealand, Papua New Guinea \"\n",
      "        },\n",
      "        \"biology\": {\n",
      "            \"description\": \" TSWV exhibits high genetic diversity (Tsompana et al., 2005; Kaye et al., 2011; Batuman et al., 2020; Fontana et al., 2020; Ruark-Seward et al., 2020). However, despite the known diversity of TSWV populations, it can be assumed that all TSWV isolates occurring worldwide are capable of causing similar diseases in plants and that differences in disease symptoms and symptom severity are a function of isolate, host and environment as well as infection time and the age of the plant (EFSA, 2012a). Diverse resistance-breaking isolates of TSWV have been reported from many regions of the world (Lopez et al., 2011; Alm\\u00e1si et al., 2020). The ability of the Tsw gene in pepper and the Sw-5 gene in tomato to break resistance to TSWV is attributed to mutations in different viral genes, NSs for Tsw/pepper (Margaria et al., 2007; de Ronde et al., 2019; Alm\\u00e1si et al., 2020) and NSm for Sw-5/tomato (Jahn et al., 2000; Hoffmann et al., 2001; Lopez et al., 2011; Olaya et al., 2020).Orthotospovirus particles are transmitted and spread in natural conditions by thrips - insects of the genera Frankliniella and Thrips (family Thripidae). Thrips species known to transmit TSWV are Frankliniella bispinosa, F. cephalica, F. fusca, F. gemina, F. occidentalis,\\u00a0F. intonsa, F. schultzei, Thrips setosus and T. tabaci (Rotenberg & Whitfield, 2018). In the EPPO region, F. occidentalis, F. intonsa\\u00a0and T. tabaci are widely distributed (https://www.cabi.org/isc/datasheet/; EFSA, 2012a). Most thrips species are very polyphagous. TSWV is transmitted by thrips in a persistent manner. Only individuals which have acquired the virus at the larval stages can transmit it. Virus transmission occurs after a latent period of circulation and multiplication of TSWV in the thrips vector. Once acquired, the virus is transmitted transstadially and thrips remain infectious for life, however there is no evidence of transovarial transmission (Ullman et al., 1993; Wijkamp et al., 1993; Wijkamp & Peters, 1993; Van de Wetering\\u00a0et al., 1996; Gupta et al., 2018; Rotenberg & Whitfield, 2018).TSWV can be spread through the movement of infected plant tissue used for vegetative propagation (EFSA, 2012a). \\u00a0Whilst the virus can also be transmitted by mechanical inoculation through small wounds on leaves or stems, this mode of transmission is unlikely to happen in the wild (EFSA, 2012a). Although\\u00a0tospoviruses are considered not to be seed-transmitted (Pappu et al., 2009), an as yet unconfirmed first report of seed transmission of a tospovirus (soybean vein necrosis virus) has recently been published by Groves et al. (2016).\"\n",
      "        },\n",
      "        \"detection and identification\": {\n",
      "            \"symptoms\": \"TSWV can cause a variety of symptoms that may vary in the same host species depending on the variety and age as well as the nutritional and environmental conditions of the plant. Symptoms of TSWV infection also vary depending on the developmental stage of the plant at the time of inoculation and on the virus isolate. Most plants respond to TSWV infection with systemic symptoms. Symptoms on the leaves of infected plants include mosaic, mottle, ring spot, and line patterns. Early infections can result in severe stunting, wilting, leaf deformation and necrosis on stems and tips, chlorotic or necrotic spots on leaves, and plant death. Symptoms on fruits usually consist of irregular discolouration, e.g. yellow/orange flecks or, occasionally, rings, or necrotic lesions or rings. Several other orthotospoviruses, as well as even more genetically distinct viruses, can cause symptoms similar to those of TSWV infection. However, it should be noted that infection of host plants by TSWV may be asymptomatic in some host plants, or the symptoms may be mild due to environmental conditions, or the infection may be too recent to reach full symptom expression (Kormelink et al., 2005; EFSA, 2012a,b; Batuman et al., 2020).\",\n",
      "            \"morphology\": \"TSWV virions are spherical, enveloped particles approximately 80-120 nm in diameter (Kormelink, 2005). The lipid envelope of orthotospoviruses contains transmembrane tips composed of two glycoproteins that form oligomeric structures on the outside of the envelope (Butkovi\\u0107 et al., 2021). The cytoplasmic tails of the glycoproteins interact with the nucleoproteins encapsulating at least one copy of the three linear ssRNA segments (Butkovi\\u0107 et al., 2021).\",\n",
      "            \"detection and inspection methods\": \"The plants, especially the leaves and fruits, should be examined for symptoms. Particular attention should be paid if thrips are present. If necessary, samples should be taken for laboratory testing for definitive identification of the pest. Procedures have been developed for the inspection of potatoes (EPPO, 2007) and vegetable plants for planting (EPPO, 2016).\"\n",
      "        },\n",
      "        \"pathways for movement\": {\n",
      "            \"description\": \" TSWV is a systemic pathogen and as such it is transmitted very efficiently by all vegetative propagation techniques. In international trade, TSWV can be transmitted by susceptible host plants for planting (whether potted or not), and is particularly able to spread if these plants also carry vectors. Thrips are easily transported on above-ground fresh plant parts (e.g. cut foliage, cut flowers and cut branches) hidden under bracts, in buds and leaf bases. Insect vectors that have acquired TSWV and that invade new areas can transmit TSWV to new hosts. Short distance spread of thrips vectors by natural means within and between adjacent greenhouses, orchards, and other production sites is likely, but long-distance spread, even in strong winds is less common. Weeds play an important role in the spread and survival of TSWV through the year; they provide a virus reservoir from which thrips vectors can migrate into crop fields, which then become heavily infected. Mechanical transmission of TSWV is inefficient in the wild and has little impact on agriculture, and transmission through seeds is considered not to be a pathway or to be a highly unlikely pathway (see Biology) (Mound, 1983; Kirk and Terry, 2003; Kormelink, 2005; EFSA, 2012a; Macharia et al., 2016; Batuman et al., 2020).\"\n",
      "        },\n",
      "        \"pest significance\": {\n",
      "            \"economic impact\": \"TSWV ranks second in the list of the ten most economically important plant viruses (Scholthof et al. 2011; Rybicki, 2015). TSWV became a major agricultural pest in the 1980s with global losses estimated at over one billion USD annually (Goldbach and Peters, 1994). The continued economic importance of TSWV results from its global distribution and wide host range, including many food and ornamental crops important to the EPPO region, significant crop losses due to infection, and the difficulty in controlling thrips and thus the virus (Adkins, 2000; Pappu et al., 2009; Scholthof et al., 2011).\",\n",
      "            \"control\": \"The use of healthy planting material and a management strategy based on a combination of thrips and weed control, the use of resistant plant varieties and other measures are considered crucial for TSWV control (EFSA, 2012a; Batuman et al., 2020).\",\n",
      "            \"phytosanitary risk\": \"TSWV is reported in many European and Mediterranean countries and has one of the largest known host ranges of any plant virus, including cultivated and wild plants. In addition, its thrips vector species are widely distributed in these countries. Spread can occur through thrips vector activity and through movement of infected hosts. The only major limiting factors are climatic conditions in Northern Europe which appear to limit the capacity of thrips vectors to develop and persist in the open environment. Since TSWV can cause both yield and quality losses in a wide range of hosts, without appropriate control measures, significant impact is expected, under protected conditions, or in open fields in Southern Europe (EFSA, 2012a).\"\n",
      "        },\n",
      "        \"phytosanitary measures\": {\n",
      "            \"description\": \" To prevent the introduction and spread of TSWV, most countries apply import requirements to some host species. These requirements vary depending on the crop and the virus prevalence at the point of origin. Following deregulation as an EU quarantine pest, TSWV was recommended for regulation as a regulated non-quarantine pest for seed potatoes, propagating and planting material (other than seeds) of some vegetables (pepper, lettuce, tomato and eggplant), for tobacco and for some ornamentals (Begonia x hiemalis, Gerbera, Pelargonium, Capsicum annuum, Chrysanthemum and Impatiens New Guinea hybrids) (Picard et al., 2018).\\u00a0EPPO countries where TSWV does not occur or is not widespread may consider regulation. If they do so, these countries may require that plants have been produced in a pest free area, in a pest-free place/site of production or be subject to inspection and/or testing (EFSA, 2012a) e.g. as performed in certification schemes. In particular, EPPO recommends that seed potatoes to be imported should meet the requirements of EPPO Standard PM 4/28 Certification scheme for seed potatoes (or equivalent) (EPPO, 1999) and be shown free from insects by visual inspection (EPPO, 2017). Post-entry quarantine programmes are established to allow the safe movement of germplasm for research and breeding purposes. During post-entry quarantine for potatoes, it is recommended that testing for viruses be carried out on microplants and/or greenhouse-grown plants; in exceptional cases, tubers (tuber sap or sprouts) may also be tested (EPPO, 2019).Before starting the propagation and/or production of plants, it should be ensured that the planting material is free from TSWV (TSWV is included in certification schemes for potatoes and various ornamental plants, see Control section). During the EU Quality pest project, in addition to the testing of nuclear stock (or the stipulation that it is derived from tested mother plants), recommended measures for all other plants for planting categories consisted of zero-tolerance based on symptom and/or testing, and active control of the vectors at the place of production.\"\n",
      "        }\n",
      "    }\n",
      "}\n"
     ]
    }
   ],
   "source": [
    "# beautiful display of dict\n",
    "import json\n",
    "print(json.dumps(datasheet_dict, indent=4))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 156,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " HOSTS 2024-06-18\n",
      " GEOGRAPHICAL DISTRIBUTION 2024-06-18\n",
      " BIOLOGY 2024-06-18\n",
      " DETECTION AND IDENTIFICATION 2024-06-18\n",
      " PATHWAYS FOR MOVEMENT 2024-06-18\n",
      " PEST SIGNIFICANCE 2024-06-18\n",
      " PHYTOSANITARY MEASURES 2024-06-18\n",
      " REFERENCES 2024-06-18\n",
      " ACKNOWLEDGEMENTS 2024-06-18\n",
      " How to cite this datasheet?\n",
      " Datasheet history 2024-06-18\n"
     ]
    }
   ],
   "source": [
    "from bs4 import BeautifulSoup\n",
    "\n",
    "# Parse the HTML\n",
    "soup = BeautifulSoup(a, 'html.parser')\n",
    "\n",
    "# Find all matching <a> tags\n",
    "links = soup.find_all('a', class_='btntitle')\n",
    "\n",
    "for link in links:\n",
    "    print(link.text)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "base",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
